catalog/build_ci/B008.md

```diff
@@ -0,0 +1,8 @@
+### B008
+
+|**Code**           | **B008** |
+|**Title**          | CI/CD agents installation and config should be in version control.|
+|**Description**    | Instuctions on how to setup the CI/CD agent(s) from scratch should be specified. i.e. there is no tacit knowledge about how to setup/install/configure an agent for the project.|
+|**Best Practices** | An interactive script at the root of repo that asks for a bare server info (e.g. ip/user/password) and then sets up runners on the specified targets automatically.|
```

Comment:
Maybe it is not the only practice or the best one.
Actually we have done another solution. We run pipelines in several dockers and put the Dockerfile in code base and also put the produced docker images to our artifact repository. So we do not need any manual configuration of the agent machine.

I suggest changing the title to:

There is no manual configuration of the CI/CD agent machines. Provisioning the required environment of the pipeline agents is derived from code base.

Reply:
Agree about the title.  
But about your above-mentioned approach, I think more parameters exist to be considered when setting up a CI agent **from scratch** (where we're not using any third-party pre-configured CI ecosystem). For example, how do you manage the docker daemon config itself? (e.g. proxy address, custom certificates, caches, max concurrent jobs, etc.)  
I've mentioned some flaws of just relying on docker images in such a case [here](https://vrgl.ir/Gd5zq).  
So I recommend describing in the item that there should be a non-manual script to set up the CI agent(s) from scratch **if not using CI as a (third-party) service**.